pyte/screens.py

```diff
@@ -614,10 +614,11 @@ def delete_lines(self, count=None):
         # If cursor is outside scrolling margins it -- do nothin'.
         if top <= self.cursor.y <= bottom:
             #                v -- +1 to include the bottom margin.
-            for _ in range(min(bottom - self.cursor.y + 1, count)):
-                self.buffer.pop(self.cursor.y)
-                self.buffer.insert(bottom, list(
-                    repeat(self.cursor.attrs, self.columns)))
+            y_offset = min(bottom, self.cursor.y + count) - self.cursor.y
+            # Shift old lines up (backwards)
+            for old_line in range(bottom, self.cursor.y + y_offset - 1, -1):
+                self.buffer[old_line - y_offset] = self.buffer[old_line]
+                self.buffer.pop(old_line)
```

Comment:
I suggest `pop(old_line, None)` in case the line you're trying to remove was all-empty, i.e. wasn't present in `self.buffer`.